extensions/amp-connatix-player/0.1/amp-connatix-player.js

```diff
@@ -71,10 +105,85 @@ export class AmpConnatixPlayer extends AMP.BaseElement {
         // Ignore messages from other iframes.
         return;
       }
-      // Player wants to close because the user interracted on its close button
-      if (getData(e) === 'cnx_close') {
-        this.destroyPlayerFrame_();
-        this.attemptCollapse();
+      const dataString = getData(e);
+      const dataJSON = tryParseJson(dataString) || dict();
+
+      if (dataJSON['event'] !== 'command') {
+        return;
+      }
+
+      switch (dataJSON['func']) {
+        // Player wants to close because the user interacted on its close button
+        case 'cnxClose': {
+          this.destroyPlayerFrame_();
+          this.attemptCollapse();
+          break;
+        }
+        // Player rendered
+        case 'cnxPlayerRendered': {
+          this.playerReadyResolver_(this.iframe_);
+          break;
+        }
+      }
+    });
+  }
+
+  /**
+   * Binds to amp-consent
+   * @private
+   */
+  bindToAmpConsent_() {
+    const consentPolicyId = super.getConsentPolicy() || 'default';
+    const consentPolicyStatePromise = getConsentPolicyState(
+      this.element,
+      consentPolicyId
+    );
+    const consentPolicyInfoPromise = getConsentPolicyInfo(
+      this.element,
+      consentPolicyId
+    );
+    const consentPolicySharedDataPromise = getConsentPolicySharedData(
+      this.element,
+      consentPolicyId
+    );
+    const consentMetadataPromise = getConsentMetadata(
+      this.element,
+      consentPolicyId
+    );
+
+    Promise.allSettled([
+      consentPolicyStatePromise,
+      consentPolicyInfoPromise,
+      consentPolicySharedDataPromise,
+      consentMetadataPromise,
+    ]).then((values) => {
+      if (values && values.length === 4) {
+        const consentPolicyState = values[0];
```

Comment:
Nit: This can be implemented with array destructuring and will result in a smaller payload in the modern builds.

`const [policyState, policyInfo, policySharedData, metadata] = values;`

Reply:
Array distructing issue on check step travis-ci

error  Array Destructuring is only allowed on known array-returning preact hooks                                                                                                                                                            local/no-array-destructuring
431  192:16  error  Replace `consentPolicyState,·consentPolicyInfo,·consentPolicySharedData,·consentMetadata` with `⏎··········consentPolicyState,⏎··········consentPolicyInfo,⏎··········consentPolicySharedData,⏎··········consentMetadata,⏎········`  prettier/prettier

Reply:
@jridgewell Down the line we should allow array destructuring in source, and transpile if we feel strongly about its performance impact. (here I'd think it's totally safe since its a single occurrence).